infoblox_client/connector.py

```diff
@@ -109,10 +114,21 @@ def _parse_options(self, options):
                 msg = "WAPI config error. Option %s is not defined" % attr
                 raise ib_ex.InfobloxConfigException(msg=msg)
 
-        for attr in ('host', 'username', 'password'):
-            if not getattr(self, attr):
-                msg = "WAPI config error. Option %s can not be blank" % attr
-                raise ib_ex.InfobloxConfigException(msg=msg)
+        def check(credentials):
+            for attr in credentials:
+                if not getattr(self, attr):
+                    msg = "WAPI config error. Option %s can not be blank" % attr
+                    raise ib_ex.InfobloxConfigException(msg=msg)
+
+        cerds_1 = {'host', 'username', 'password'}
+        cerds_2 = {'host', 'cert', 'key'}
+        if cerds_1.issubset(options):
```

Comment:
If both certificate and username/pass are provided, the client would prefer username and password. Is this behavior expected?

Reply:
Aren't we running the same checks in here? Set method `issubset` will check if provided `options` set is subset of the creds_1. `Check` method will basically make the same check.

Reply:
yes, if both values are provided it will prefer username and password.